File: android/app/src/main/kotlin/dev/tcode/thinmpf/api/PlayerFlutterApiImpl.kt
package dev.tcode.thinmpf.api

import dev.tcode.thinmpf.extension.toPigeon
import dev.tcode.thinmpf.model.contract.SongModelContract
import dev.tcode.thinmpf.pigeon_output.PlayerFlutterApi
import io.flutter.plugin.common.BinaryMessenger

class PlayerFlutterApiImpl {
    companion object {
        lateinit var api: PlayerFlutterApi
        fun setUp(binaryMessenger: BinaryMessenger) {
            api = PlayerFlutterApi(binaryMessenger)
        }
    }

    fun onIsPlayingChange(isPlaying: Boolean) {
        api.onIsPlayingChange(isPlaying) { }
    }

    fun onPlaybackSongChange(song: SongModelContract) {
        api.onPlaybackSongChange(song.toPigeon()) { }
    }
}

File: ios/Runner/Api/PlayerFlutterApiImpl.swift
//
//  PlayerFlutterApiImpl.swift
//  Runner
//
//  Created by tk on 2024/07/28.
//

class PlayerFlutterApiImpl {
    static var api: PlayerFlutterApi!

    static func setUp(binaryMessenger: FlutterBinaryMessenger) {
        api = PlayerFlutterApi(binaryMessenger: binaryMessenger)
    }

    func onIsPlayingChange(isPlaying: Bool, completion: @escaping (Result<Void, PigeonError>) -> Void = { _ in }) {
        PlayerFlutterApiImpl.api.onIsPlayingChange(isPlaying: isPlaying, completion: completion)
    }

    func onPlaybackSongChange(song: SongModelContract, completion: @escaping (Result<Void, PigeonError>) -> Void = { _ in }) {
        PlayerFlutterApiImpl.api.onPlaybackSongChange(song: song.toPigeon(), completion: completion)
    }
    
    func onError(message: String, completion: @escaping (Result<Void, PigeonError>) -> Void = { _ in }) {
        PlayerFlutterApiImpl.api.onError(message: message, completion: completion)
    }
}
